monthly-contributions/src/main/scala/com/gu/support/workers/lambdas/ContributionCompleted.scala

```diff
@@ -35,8 +35,15 @@ class ContributionCompleted
     )
   }
 
-  def putContributionCompleted(paymentMethod: String): Future[Unit] =
-    new RecurringContributionsMetrics(paymentMethod.toLowerCase, "monthly")
+  def putContributionCompleted(paymentMethod: String): Future[Unit] = {
+    val paymentMethodLabel = {
```

Comment:
A better way to do this (because it doesn't rely on string matches and will be caught by the compiler if the classname changes) is to use a match on the classtype:

      def putContributionCompleted(paymentMethod: PaymentMethod): Future[Unit] = {
        val paymentMethodLabel = paymentMethod match {
          case CreditCardReferenceTransaction => "stripe"
          case PaypalPaymentReferenceTransaction => "paypal"
        }
        new RecurringContributionsMetrics(paymentMethodLabel, "monthly")
          .putContributionCompleted().recover({ case _ => () })
      }

Also I think it would be good to refactor the RecurringContributionsMetrics class so that it takes a PaymentMethod object rather than a string, then you can do the conversion there (actually in the ProductDimensions class probably) rather than in the calling class and you'll only need to write code for it once.

Reply:
That sounds like a much more elegant solution. Thanks!